### PR TITLE
Added support for Identify Cluster

### PR DESCRIPTION
**Pull request by kkasperczyk-no**
#### Problem
Weather station app doesn't support Identify cluster.

#### Change overview
* Enabled Identify cluster in the `weather-station.zap` and regenerated source code files
* Added `buzzer.cpp` file allowing managing the buzzer
* Implemented signalling the identification process using the buzzer

### CODE REVIEW

File: examples/weather_station/src/app_task.cpp [hunk 1]
```diff
@@ -67,6 +73,12 @@ bool sHaveBLEConnections;
 
 LedState sLedState = LedState::kAlive;
 
+static uint16_t sIdentifyDurationMs;
```
Comment (Damian-Nordic) [inline examples/weather_station/src/app_task.cpp:76]: static is unneeded as these variables are already in anonymous namespace.

File: examples/weather_station/src/buzzer.cpp [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+}
+
+void BuzzerSetState(bool onOff) {
+    sBuzzerState = onOff;
+    pwm_pin_set_usec(kBuzzerDevice, BUZZER_PWM_PIN, kBuzzerPwmPeriodUs, sBuzzerState ? kBuzzerPwmDutyCycleUs : 0,
+				       BUZZER_PWM_FLAGS);
+}
+
+void BuzzerToggleState() {
+    sBuzzerState = !sBuzzerState;
```
Comment (Damian-Nordic) [inline examples/weather_station/src/buzzer.cpp:38]: `BuzzerSetState(!sBuzzerState)`?

File: examples/weather_station/src/buzzer.cpp [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+#define BUZZER_PWM_NODE DT_ALIAS(buzzer_pwm)
+#define BUZZER_PWM_NAME DT_LABEL(BUZZER_PWM_NODE)
+#define BUZZER_PWM_PIN DT_PROP(BUZZER_PWM_NODE, ch0_pin)
+#define BUZZER_PWM_FLAGS DT_PWMS_FLAGS(BUZZER_PWM_NODE)
+
+constexpr uint16_t kBuzzerPwmPeriodUs = 10000;
+constexpr uint16_t kBuzzerPwmDutyCycleUs = 5000;
+
+const device *kBuzzerDevice;
```
Comment (Damian-Nordic) [inline examples/weather_station/src/buzzer.cpp:19]: It's not constant - don't use `k` prefix. Also, please put both in anonymous namespace or mark both as static constistently.

File: examples/weather_station/src/app_task.cpp [hunk 4]
```diff
@@ -254,6 +277,47 @@ void AppTask::MeasurementsTimerHandler()
 	sAppTask.UpdateClusterState();
 }
 
+void AppTask::OnIdentifyStart(Identify *)
+{
+	uint16_t identifyTimeS;
+	EmberAfStatus status = emberAfReadAttribute(kIdentifyEndpointId, ZCL_IDENTIFY_CLUSTER_ID, ZCL_IDENTIFY_TIME_ATTRIBUTE_ID, CLUSTER_MASK_SERVER, reinterpret_cast<uint8_t *>(&identifyTimeS), sizeof(identifyTimeS), NULL);
```
Comment (Damian-Nordic) [inline examples/weather_station/src/app_task.cpp:283]: Use `Identify:Attributes::IndentifyTime::Get` accessor?

File: examples/weather_station/src/app_task.cpp [hunk 5]
```diff
@@ -254,6 +277,47 @@ void AppTask::MeasurementsTimerHandler()
...
+	if (status != EMBER_ZCL_STATUS_SUCCESS) {
+		LOG_ERR("Reading identify time failed %x", status);
+		return;
+	}
+	sIdentifyFullDurationTimeMs = identifyTimeS * 1000;
```
Comment (Damian-Nordic) [inline examples/weather_station/src/app_task.cpp:288]: Do we need to maintain that? Isn't `OnIdentifyStop` called when the timeout expires?

File: examples/weather_station/src/app_task.cpp [hunk 6]
```diff
@@ -254,6 +277,47 @@ void AppTask::MeasurementsTimerHandler()
...
+		status = emberAfWriteAttribute(kIdentifyEndpointId, ZCL_IDENTIFY_CLUSTER_ID,
+					       ZCL_IDENTIFY_TIME_ATTRIBUTE_ID, CLUSTER_MASK_SERVER,
+					       reinterpret_cast<uint8_t *>(&identifyTime), ZCL_INT16U_ATTRIBUTE_TYPE);
+
+		if (status != EMBER_ZCL_STATUS_SUCCESS) {
+			LOG_ERR("Updating identify time failed %x", status);
+		}
+
+		if (identifyTime == 0) {
+			k_timer_stop(&sIdentifyTimer);
```
Comment (Damian-Nordic) [inline examples/weather_station/src/app_task.cpp:314]: I think `OnIdentifyStop` should call that and this function should simply `toggle the buzzer state`. Wouldn't it be enough?

File: examples/weather_station/src/app_task.cpp (hunk 4 above)
Comment (kkasperczyk-no) [reply examples/weather_station/src/app_task.cpp]: Ok, in that case I also changed it for the measurements clusters

File: examples/weather_station/src/app_task.cpp (hunk 5 above)
Comment (kkasperczyk-no) [reply examples/weather_station/src/app_task.cpp:291]: It is, but this variable is used to update cluster state every second with time left to the identify end

File: examples/weather_station/src/app_task.cpp (hunk 6 above)
Comment (kkasperczyk-no) [reply examples/weather_station/src/app_task.cpp]: > I think OnIdentifyStop should call that and this function

Right, done

> this function should simply toggle the buzzer state. Wouldn't it be enough?

What do you mean by this function? `OnIdentifyStop` or the handler? Anyway buzzer needs to be toggled in the handler, as it beeps in the 500 ms on - 500 ms off cycle for the identify time.

Comment (Damian-Nordic) [reply examples/weather_station/src/app_task.cpp]: I mean that the timer handler should just toggle the buzzer and it shouldn't care if the indentify is enabled. `OnIdentifyStop` will reset the timer when the handler should no longer be called.

Comment (kkasperczyk-no) [reply examples/weather_station/src/app_task.cpp]: @Damian-Nordic yeah, I tested that indeed cluster state is updated automatically by the core, so removed all of the code handling it.

**Review by Damian-Nordic — APPROVED** [review]

**Pull request merged**